### ISSUE TITLE
How to add regularizer to a specific layer outside model.py?

### ISSUE THREAD

**Issue by keineahnung2345** [matterport/Mask_RCNN#1013]
When I trained my model, I found that `val_rpn_bbox_loss` tend to overfit easily, so I try to add the regularizer to the layer `rpn_bbox_pred` in my train_xxx.ipynb, 
```
model.keras_model.inner_model.get_layer(name="rpn_model").get_layer("rpn_bbox_pred"). \
kernel_regularizer = regularizers.l2(200) #large value just for experiment
```
But when I take a look at the loss in training, I found that the loss remains at the same scale as before.

(In contrast, I also experimented by adding regularizer into `rpn_bbox_pred` layer in model.py, and this time, the loss becomes more than ten thousands, just as expected.)

I later found that a model should be recompiled so that the changes will take effect. When I checked https://github.com/matterport/Mask_RCNN/blob/master/mrcnn/model.py#L2186, it seems that you must get losses before compiling a model, but at the time it calculates losses, the model hasn't been recompiled, so the regularizer I have added outside model.py won't be taken into consideration.

So  is there a workaround that can solve or bypass the problem? 
Thanks in advance.
